src/configs/gradle.rs

```diff
@@ -0,0 +1,73 @@
+use std::fmt::Display;
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Deserialize, Serialize)]
+#[cfg_attr(
+    feature = "config-schema",
+    derive(schemars::JsonSchema),
+    schemars(deny_unknown_fields)
+)]
+#[serde(default)]
+pub struct GradleConfig<'a> {
+    pub format: &'a str,
+    pub version_format: &'a str,
+    pub symbol: &'a str,
+    pub style: &'a str,
+    pub disabled: bool,
+    pub detect_extensions: Vec<&'a str>,
+    pub detect_files: Vec<&'a str>,
+    pub detect_folders: Vec<&'a str>,
+    pub strategy: GradleVersionStrategy,
+}
+
+#[derive(Clone, Deserialize, Serialize)]
+#[cfg_attr(
+    feature = "config-schema",
+    derive(schemars::JsonSchema),
+    schemars(deny_unknown_fields)
+)]
+#[serde(try_from = "String")]
+pub enum GradleVersionStrategy {
+    #[serde(rename = "properties")]
+    WrapperProperties,
+    #[serde(rename = "executable")]
+    Executable,
+}
```

Comment:
Instead of renaming every option, please use `rename_all=pascal_case`:

```suggestion
#[serde(try_from = "String")]
pub enum GradleVersionStrategy {
    #[serde(rename = "properties")]
    WrapperProperties,
    #[serde(rename = "executable")]
    Executable,
}
```

Reply:
Yes I meant that.